test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -119,6 +119,17 @@ def run(self, high_level_object, **options):
         return qc
 
 
+class OpAPluginNeedsCouplingMap(HighLevelSynthesisPlugin):
+    """Synthesis plugins for OpA that needs a coupling map to be run."""
+
+    def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **options):
+        if coupling_map is None:
+            raise TranspilerError("Coupling map should be specified!")
```

Comment:
I guess it doesn't matter in this case since we're just testing the raise, but  you can do `target.build_coupling_map()` to get a coupling map from a target.

Reply:
Yes, good to know, thanks!

Reply:
I have also added a test that when only the target is set, the synthesis plugin does see a valid coupling map (because indeed the `HighLevelSynthesis` pass builds it using the function above).